File: 코딩테스트 연습 - 크레인 인형뽑기 게임 _ 프로그래머스.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for(int move : moves) {
            for(int i = 0; i < board.length; i++) {
                if(board[i][move-1] != 0) {
                    if(!stack.isEmpty()){
                        if (stack.peek() == board[i][move-1]){
                            stack.pop();
                            answer++;
                        }else {
                            stack.push(board[i][move-1]); 
                        }
                    }else {
                        stack.push(board[i][move-1]);
                    }
                    board[i][move-1] = 0;
                    break;
                }
            }
        }
        return answer*2;
    }
}

File: python/코딩테스트 연습 - 크레인 인형뽑기 게임 _ 프로그래머스.py
def solution(board, moves):
    answer = 0
    stk = []
    for m in moves:
        for i in range(len(board)):
            if board[i][m-1] != 0:
                stk.append(board[i][m-1])
                board[i][m-1] = 0
                
                if len(stk) > 1:
                    if stk[-1] == stk[-2]:
                        stk.pop(-1)
                        stk.pop(-1)
                        answer += 2
                break
    return answer